File: Docs/ScriptApi/Examples/QueryMonthlySummary.py
def query_main(context):
    totals = {}
    for item in context.diary.workItems.stream(pageSize=500, range="thisMonth"):
        tag_name = "无"
        for tag in item.get("tags", []):
            if tag.get("isPrimary", False):
                tag_name = tag.get("name") or "无"
                break
        totals[tag_name] = totals.get(tag_name, 0.0) + item.get("hours", 0.0)

    lines = ["主标签 | 工时"]
    for name, hours in sorted(totals.items(), key=lambda pair: pair[1], reverse=True):
        lines.append(f"{name} | {hours:.2f} 小时")

    context.log.info("\n".join(lines))
    return None

File: Docs/ScriptApi/Examples/QueryMonthlySummary.lua
function query_main(context)
    local totals = {}
    for item in diary.work_items.stream({ range = "thisMonth", pageSize = 500 }) do
        local tagName = "无"
        for _, tag in ipairs(item.tags) do
            if tag.isPrimary then
                tagName = tag.name
                break
            end
        end
        totals[tagName] = (totals[tagName] or 0) + item.hours
    end

    local names = {}
    for name in pairs(totals) do
        table.insert(names, name)
    end
    table.sort(names, function(a, b) return totals[a] > totals[b] end)

    local lines = { "主标签 | 工时" }
    for _, name in ipairs(names) do
        table.insert(lines, string.format("%s | %.2f 小时", name, totals[name]))
    end

    diary.log.info(table.concat(lines, "\n"))
end
